test/extended/builds/volumes.go

```diff
@@ -126,3 +133,233 @@ var _ = g.Describe("[sig-builds][Feature:Builds][volumes] build volumes", func()
 		})
 	})
 })
+
+var _ = g.Describe("[sig-builds][Feature:Builds][volumes] csi build volumes", func() {
+	var (
+		oc                     = exutil.NewCLI("build-volumes")
+		oc1                    = exutil.NewCLI("build-volumes-1")
+		baseDir                = exutil.FixturePath("testdata", "builds", "volumes")
+		secret                 = filepath.Join(baseDir, "secret.yaml")
+		s2iDeploymentConfig    = filepath.Join(baseDir, "s2i-deploymentconfig.yaml")
+		s2iImageStream         = filepath.Join(baseDir, "s2i-imagestream.yaml")
+		dockerDeploymentConfig = filepath.Join(baseDir, "docker-deploymentconfig.yaml")
+		dockerImageStream      = filepath.Join(baseDir, "docker-imagestream.yaml")
+		// csi enabled volume specifics
+		csiSharedSecret                            = filepath.Join(baseDir, "csi-shared-secret.yaml")
+		csiSharedRole                              = filepath.Join(baseDir, "csi-sharedresourcerole.yaml")
+		csiSharedRoleBinding                       = filepath.Join(baseDir, "csi-sharedresourcerolebinding.yaml")
+		csiS2iBuildConfig                          = filepath.Join(baseDir, "csi-s2i-buildconfig.yaml")
+		csiDockerBuildConfig                       = filepath.Join(baseDir, "csi-docker-buildconfig.yaml")
+		csiWihthoutResourceRefreshS2iBuildConfig   = filepath.Join(baseDir, "csi-without-rr-s2i-buildconfig.yaml")
+		csiWithoutResourceRefreshDockerBuildConfig = filepath.Join(baseDir, "csi-without-rr-docker-buildconfig.yaml")
+		techPreviewNoUpgrade                       = false
+	)
+
+	g.Context("", func() {
+		g.BeforeEach(func() {
+			exutil.PreTestDump()
+		})
+
+		g.JustBeforeEach(func() {
+			noUpgrade, err := isNoUpgrade(oc.AdminConfig())
+			if err != nil {
+				e2e.Failf("failed to get dynamic client: %v", err)
+			}
+			techPreviewNoUpgrade = noUpgrade
+		})
+
+		g.AfterEach(func() {
+			if g.CurrentGinkgoTestDescription().Failed {
+				exutil.DumpPodStates(oc)
+				exutil.DumpConfigMapStates(oc)
+				exutil.DumpPodLogsStartingWith("", oc)
+			}
+		})
+
+		if !techPreviewNoUpgrade {
```

Comment:
I _don't_ think this will work the way you expect it to. I'm pretty positive this breaks how openshift-tests works with ginkgo.

I would use skips in each test instead, and/or separate context/Describe sections for tech preview vs. standard cluster.

See https://github.com/openshift/origin/blob/master/test/extended/README.md

Reply:
Thank you @adambkaplan , Good catch
Fixed now